### PR TITLE
ntp: handle drag + drop outside of favorites

### PR DESCRIPTION
**Pull request by shakyShane**
**Asana Task/Github Issue:** https://app.asana.com/0/1201141132935289/1208717262504806/f

## Description

- [x] We need to prevent responding to arbitrary drops, otherwise the browser pops open a new window
- [x] Added a graceful way to handle fatal exceptions (we now display them in the page for debugging/integration)

## Testing Steps

- ensure drag+drop still works here https://deploy-preview-1256--content-scope-scripts.netlify.app/build/pages/new-tab/
- locally, throw some exception inside the `init` function in `special-pages/pages/new-tab/app/index.js` - ensure you get some debug output.

## Checklist

*Please tick all that apply:*

- [x] I have tested this change locally
- [x] I have tested this change locally in all supported browsers
- [ ] This change will be visible to users
- [ ] I have added automated tests that cover this change
- [ ] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

### CODE REVIEW

File: special-pages/index.mjs [hunk 1]
```diff
@@ -133,12 +133,15 @@ for (const buildJob of buildJobs) {
...
+        const output = buildSync({
             entryPoints: buildJob.entryPoints,
             outdir: buildJob.outputDir,
             bundle: true,
-            format: 'iife',
+            // metafile: true,
+            // minify: true,
+            // splitting: true,
             // external: ['../assets/img/*'],
+            format: 'esm',
```
Comment (shakyShane) [inline special-pages/index.mjs:144]: leaving these here for debugging - soon we'll want the splitting support.

File: special-pages/index.mjs [hunk 2]
```diff
@@ -156,6 +159,10 @@ for (const buildJob of buildJobs) {
             },
             dropLabels: buildJob.injectName === 'integration' ? [] : ['$INTEGRATION'],
         });
+        if (output.metafile) {
+            const meta = JSON.stringify(output.metafile, null, 2);
+            writeFileSync(join(buildJob.outputDir, 'metafile.json'), meta);
+        }
```
Comment (shakyShane) [inline special-pages/index.mjs:165]: this produces an output file you can drop into https://esbuild.github.io/analyze/ - it was useful for me debugging why the auto-animate library was being pulled into the NTP

File: special-pages/pages/new-tab/src/js/index.js [hunk 3]
```diff
@@ -75,8 +76,41 @@ const rawMessaging = createSpecialPageMessaging({
...
+                </p>
+                <br />
+                <pre style={{ whiteSpace: 'prewrap', overflow: 'auto', fontSize: '.8em' }}>
+                    <code>{JSON.stringify(telemetry.eventStore, null, 2)}</code>
+                </pre>
+            </div>
+        </Fragment>
+    );
+    render(element, document.body);
 });
```
Comment (shakyShane) [inline special-pages/pages/new-tab/src/js/index.js:116]: This is deliberately outside of the `app` folder - if we get here it's unrecoverable, so we just want some clean output for those doing the integration

File: special-pages/pages/new-tab/app/index.js
Comment (mgurgel) [inline special-pages/pages/new-tab/app/index.js:1]: Nice cleanup!

**Review by mgurgel — DISMISSED** [review]
👍

**Pull request merged**